src/encord_active/lib/coco/encoder.py

```diff
@@ -260,29 +261,21 @@ def get_image(self, label_hash: str, data_unit: dict) -> dict:
             "flickr_url": "",
             "id": image_id,
             "image_title": data_unit["data_title"],
-            "file_name": self.get_file_name_and_download_image(label_hash, data_unit),
+            "file_name": self.get_local_path_or_file_name(data_unit),
             "height": data_unit["height"],
             "width": data_unit["width"],
             "label_hash": label_hash,
             "data_hash": data_hash,
             "frame_num": int(data_unit["data_sequence"]),
         }
 
-    def get_file_name_and_download_image(self, label_hash: str, data_unit: dict) -> str:
-        data_hash = data_unit["data_hash"]
-        url = data_unit["data_link"]
-        image_extension = data_unit["data_type"].split("/")[-1]
-
-        relative_destination_path = Path("data").joinpath(
-            Path(label_hash), Path("images"), Path(f"{data_hash}.{image_extension}")
-        )
-        absolute_destination_path = self._download_file_path.joinpath(relative_destination_path)
-        download_condition = self._download_files and not absolute_destination_path.exists()
-
-        if download_condition or self._force_download:
-            self.download_image(url, absolute_destination_path)
-
-        return str(relative_destination_path)
+    def get_local_path_or_file_name(self, data_unit: dict) -> str:
+        url: str = data_unit["data_link"]
```

Comment:
The signed url in data_link has a risk of being out of date: using project_data_structure.label_structure(label_hash).iter_data_units(data_unit_hash=du_hash).signed_url

this will return the latest cached value from the caching infra, which should be more likely to be up to date. (though for long running streamlit instances this can also be out-of-date - though coco export would be the only part that this effects).

Reply:
I just removed the link. That will remove confusion.